src/DIRAC/FrameworkSystem/Agent/MonitoringAgent.py

```diff
@@ -0,0 +1,544 @@
+        res = self.sysAdminClient.stopComponent(system, name)
+        if not res['OK']:
+          self.logError('Failed to stop component:', '%s: %s' % (instance, res['Message']))
+        else:
+          self.accounting[instance]['Treatment'] = 'Instance was running, stopped instance'
+      else:
+        self.accounting[instance]['Treatment'] = 'Instance is running, should be stopped'
+
+  def checkURLs(self):
+    """Ensure that the running services have their URL in the Config."""
+    self.log.info('Checking URLs')
+    # get services again, in case they were started/stop in controlComponents
+    gConfig.forceRefresh(fromMaster=True)
+    res = self.getRunningInstances(instanceType='Services', runitStatus='All')
+    if not res['OK']:
+      return S_ERROR('Failure to get running services')
+    self.services = res['Value']
+    for service, options in self.services.iteritems():
+      self.log.debug('Checking URL for %s with options %s' % (service, options))
+      # ignore SystemAdministrator, does not have URLs
+      if 'SystemAdministrator' in service:
+        continue
+      self._checkServiceURL(service, options)
+
+    if self.csAPI.csModified and self.commitURLs:
+      self.log.info('Commiting changes to the CS')
+      result = self.csAPI.commit()
+      if not result['OK']:
+        self.logError('Commit to CS failed', result['Message'])
+        return S_ERROR('Failed to commit to CS')
+    return S_OK()
+
+  def _checkServiceURL(self, serviceName, options):
+    """Ensure service URL is properly configured in the CS."""
+    url = self._getURL(serviceName, options)
+    system = options['System']
+    module = options['Module']
+    self.log.info('Checking URLs for %s/%s' % (system, module))
+    urlsConfigPath = os.path.join('/Systems', system, self.setup, 'URLs', module)
+    urls = gConfig.getValue(urlsConfigPath, [])
+    self.log.debug('Found configured URLs for %s: %s' % (module, urls))
+    self.log.debug('This URL is %s' % url)
+    runitStatus = options['RunitStatus']
+    wouldHave = 'Would have ' if not self.commitURLs else ''
+    if runitStatus == 'Run' and url not in urls:
+      urls.append(url)
+      message = '%sAdded URL %s to URLs for %s/%s' % (wouldHave, url, system, module)
+      self.log.info(message)
+      self.accounting[serviceName + '/URL']['Treatment'] = message
+      self.csAPI.modifyValue(urlsConfigPath, ','.join(urls))
+    if runitStatus == 'Down' and url in urls:
+      urls.remove(url)
+      message = '%sRemoved URL %s from URLs for %s/%s' % (wouldHave, url, system, module)
+      self.log.info(message)
+      self.accounting[serviceName + '/URL']['Treatment'] = message
+      self.csAPI.modifyValue(urlsConfigPath, ','.join(urls))
+
+  @staticmethod
+  def _getURL(serviceName, options):
+    """Return URL for the service."""
+    system = options['System']
+    port = options['Port']
+    host = socket.gethostname()
+    url = 'dips://%s:%s/%s/%s' % (host, port, system, serviceName)
```

Comment:
Anything to do here for HTTPS?

Reply:
probably :)

Reply:
It's actually a bit tricky since you craft the URL by hand. Maybe you could change a bit the approach. Since you anyway have this `_checkServiceURL` check, why not retrieving the list of URL from the CS, and either reporting an error `host:port/System/Service` is not there, or returning the full URL ? 
The alternative is that in your `getRunningInstances` method, since you check the service configuration in the CS, you can fetch the `Protocol` flag. If it is not `https` you use `dips`

Reply:
If the difference is only the protocol, I can pick that up from the options. 

But in the certification for the filecatalog, I don't see the "Port" in the options for that services, in the URL the port has 8443. Is that the default and I can assume that for https unless set otherwise, or is that set centrally somewhere?

Reply:
No, `Port` is taken from the `Tornado` config. It indeed is the default

Reply:
I added the picking up of the port from Tornado/Services and stuff